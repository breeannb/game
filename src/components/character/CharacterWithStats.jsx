import React from 'react';
import PropTypes from 'prop-types';
import Character from './Character';
import Stats from '../stats/Stats';

const CharacterWithStats = ({ character }) => (
  
  <section>
    This is the StatsCharacter page
    <Character image={character.image} />
    <Stats
      speed={character.speed}
      might={character.might}
      sanity={character.sanity}
      knowledge={character.knowledge}
    />
  </section>
);

CharacterWithStats.propTypes = {
  character: PropTypes.shape({
    image: PropTypes.string.isRequired,
    speed: PropTypes.number.isRequired,
    might: PropTypes.number.isRequired,
    sanity: PropTypes.number.isRequired,
    knowledge: PropTypes.number.isRequired
  })
};

export default CharacterWithStats;
