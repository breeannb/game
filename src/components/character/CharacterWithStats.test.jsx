import React from 'react';
import { shallow } from 'enzyme';
import CharacterWithStats from './CharacterWithStats';

describe('CharacterWithStats component', () => {
  it('renders CharacterWithStats', () => {
    const character = {
      image: 'imageOne.jpeg',
      speed: 3,
      might: 4,
      sanity: 10,
      knowledge: 3
    };
    const wrapper = shallow(<CharacterWithStats character={character} />);
    expect(wrapper).toMatchSnapshot();
  });
});
