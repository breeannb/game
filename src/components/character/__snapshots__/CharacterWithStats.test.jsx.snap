// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterWithStats component renders CharacterWithStats 1`] = `
<section>
  <Character
    image="imageOne.jpeg"
  />
  <Stats
    knowledge={3}
    might={4}
    sanity={10}
    speed={3}
  />
</section>
`;
